system-test/system_test.sh

```diff
@@ -12,13 +12,28 @@ retry() {
 
 cd $(dirname $0)
 
+if [[ -z "$BINARY_HOST" ]]; then
+  BUILD_BINARIES="true"
+fi
+
 for i in 6 8 10 11; do
   # Test Linux support for the given node version.
-  retry docker build -f Dockerfile.linux --build-arg NODE_VERSION=$i -t node$i-linux .
-  docker run -v $PWD/..:/src node$i-linux /src/system-test/test.sh
+  retry docker build \
+  -f Dockerfile.linux \
+  --build-arg NODE_VERSION=$i --build-arg BUILD_BINARIES="$BUILD_BINARIES" \
+  -t node$i-linux .
+
+ docker run -v $PWD/..:/src \
+  -e BINARY_HOST="$BINARY_HOST" \
+  node$i-linux /src/system-test/test.sh
+
   # Test Alpine support for the given node version.
-  docker build -f Dockerfile.node$i-alpine -t node$i-alpine .
-  docker run -v $PWD/..:/src node$i-alpine /src/system-test/test.sh
+  retry docker build -f Dockerfile.node$i-alpine \
+    --build-arg BUILD_BINARIES="$BUILD_BINARIES" \
+    -t node$i-alpine .
+
+  docker run -v $PWD/..:/src -e BINARY_HOST="$BINARY_HOST" \
```

Comment:
Nit: `-e BINARY_HOST="$BINARY_HOST"` is same line here but separate line in the docker run command above.

Reply:
Tried to be more consistent (went for  separating lines to keep lines less than 80 characters)